### PR TITLE
Fix spelling

### PR DESCRIPTION
**Pull request by PeterPetrik**
only remaining bits are in `QgsSettings` where we use "browser/favourites" (follow up for https://github.com/qgis/QGIS/pull/33679#issuecomment-596535820)

### CODE REVIEW

**Comment by m-kuhn** [comment]
Looks good.
Can you add it to scripts/spell_checks/spelling.dat too?

**Review by elpaso — APPROVED** [review]

File: src/core/qgsbrowserproxymodel.h [hunk 1]
```diff
@@ -147,7 +147,7 @@ class CORE_EXPORT QgsBrowserProxyModel : public QSortFilterProxyModel
      * Sets a filter to hide data items based on on item's data item provider key.
```
Comment (DelazJ) [inline src/core/qgsbrowserproxymodel.h:147]:
```suggestion
     * Sets a filter to hide data items based on item's data item provider key.
```
and is it **item**'s data **item** provider key?

File: src/core/qgsbrowserproxymodel.h [hunk 2]
```diff
@@ -165,7 +165,7 @@ class CORE_EXPORT QgsBrowserProxyModel : public QSortFilterProxyModel
      * Sets a filter to show data items based on on item's data item provider key.
```
Comment (DelazJ) [inline src/core/qgsbrowserproxymodel.h:165]: ditto

**Review by m-kuhn — APPROVED** [review]
Nice work :tada:

File: src/core/layout/qgslayoutexporter.h [hunk 3]
```diff
@@ -336,7 +336,7 @@ class CORE_EXPORT QgsLayoutExporter
        *
        * \since QGIS 3.14
        */
-      bool exportLayersAsSeperateFiles = false;
+      bool exportLayersAsSeparateFiles = false;
```
Comment (PeterPetrik) [inline src/core/layout/qgslayoutexporter.h:339]: @nyalldawson this is basically API change.. do we bother?

Comment (nyalldawson) [reply src/core/layout/qgslayoutexporter.h:339]: Bah -- better add a "TODO QGIS 4" note and leave it at that...

**Pull request merged**